### DISCUSSION TITLE
Need help to configure the script file in Azure DevOps

### DISCUSSION THREAD

**Discussion by arunkumar5358** [actions/virtual-environments#3816] [Image configuration]
Hi,

I have created a json file and used the below value in the template, i am using build immutable option to create the image using azure pipeline, but i am getting the below error, i am not sure what is the right way to use the template, can some please help me to rectify the issue? It would be good if someone explains clearly where to configure the variables such as {{template_dir}}, {{ .Vars }} {{ .Path }}'

       {
            "type": "shell",
            "script": "{{template_dir}}/scripts/base/apt-mock.sh",
            "execute_command": "sudo sh -c '{{ .Vars }} {{ .Path }}'"
        },

"* Bad script '/agent/_work/1/s/packer/scripts/base/apt-mock.sh': stat /agent/_work/1/s/packer/scripts/base/apt-mock.sh: no such file or directory"

**Comment by al-cheb** [comment]
Hi, @arunkumar5358 
`template_dir` - The directory to the template for the build (https://www.packer.io/docs/templates/legacy_json_templates/engine). 

`execute_command (string)`(https://www.packer.io/docs/provisioners/shell) - The command to use to execute the script. By default this is `chmod +x {{ .Path }}; {{ .Vars }} {{ .Path }}`, unless the user has set "use_env_var_file": true -- in that case, the default execute_command is `chmod +x {{.Path}}; . {{.EnvVarFile}} && {{.Path}}`. This is a template engine. Therefore, you may use user variables and template functions in this field. In addition, there are three available extra variables:

`Path` is the path to the script to run
`Vars` is the list of environment_vars, if configured.
`EnvVarFile` is the path to the file containing env vars, if use_env_var_file is true.


For example:
packer build .... `C:\agent\_work\1\s\images\linux\ubuntu1804.json`
`template_dir` = `C:\agent\_work\1\s\images\linux`

```
{
            "type": "shell",
            "scripts": [
                "{{template_dir}}/scripts/installers/docker-compose.sh",
            ],
            "environment_vars": [
                "DOCKERHUB_LOGIN={{user `dockerhub_login`}}",
                "DOCKERHUB_PASSWORD={{user `dockerhub_password`}}"
            ],
            "execute_command": "sudo sh -c '{{ .Vars }} {{ .Path }}'"
        }
```

 "execute_command": "sudo sh -c '{{ .Vars }} {{ .Path }}'"  converts into `sudo sh -c 'DOCKERHUB_LOGIN=test_login DOCKERHUB_PASSWORD=test_password C:\agent\_work\1\s\images\linux/scripts/installers/docker-compose.sh"`

**Reply by arunkumar5358** [reply]
Hi Aleksandr Chebotov, 

Thank you so much for your explanation! let me try and come back if anything else is required..
